src/nativestore/NodeManager.cpp

```diff
@@ -264,9 +305,12 @@ void NodeManager::addNodeIndex(std::string nodeId, unsigned int nodeIndex) {
         char nodeIDC[NodeManager::INDEX_KEY_SIZE] = {0};  // Initialize with null chars
         std::strcpy(nodeIDC, nodeId.c_str());
         index_db.write(nodeIDC, sizeof(nodeIDC));
-        index_db.write(reinterpret_cast<char *>(&(nodeIndex)), sizeof(unsigned int));
-        node_manager_logger.debug("Writing node index --> Node key = " + std::string(nodeIDC) + " value " +
-                                  std::to_string(nodeIndex));
+        index_db.write(reinterpret_cast<char *>(&nodeIndex), sizeof(unsigned int));
+        index_db.flush();
+        node_manager_logger.debug("Writing node index --> Node key = " +
```

Comment:
```suggestion
        node_manager_logger.info("Writing node index --> Node key = " +
```
Debug logs only appear if debug mode is enabled. I thinks its better to be `info` here

Reply:
They should not be logged in production mode.

Reply:
I also believe we should keep this log at debug level. Info level logging is unnecessarily verbose.